modules/s3/main.tf

```diff
@@ -241,6 +241,39 @@ resource "aws_apigatewayv2_route" "post_request" {
   target    = "integrations/${aws_apigatewayv2_integration.post_request.id}"
 }
 
+### Requests API - Update request
+resource "aws_apigatewayv2_route" "update_request" {
+  api_id = aws_apigatewayv2_api.ugt_gw.id
+
+  route_key = "PUT /api/{version}/requests/{id}"
+  target    = "integrations/${aws_apigatewayv2_integration.post_request.id}"
+}
+
+### Requests API - List requests
+resource "aws_lambda_permission" "list_requests" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.list_requests.function_name
+  principal     = "apigateway.amazonaws.com"
+
+  source_arn = "${aws_apigatewayv2_api.ugt_gw.execution_arn}/*/*"
+}
+
+resource "aws_apigatewayv2_integration" "list_requests" {
+  api_id = aws_apigatewayv2_api.ugt_gw.id
+
+  integration_uri    = aws_lambda_function.list_requests.invoke_arn
+  integration_type   = "AWS_PROXY"
+  integration_method = "POST"
+}
+
+resource "aws_apigatewayv2_route" "list_requests" {
+  api_id = aws_apigatewayv2_api.ugt_gw.id
+
+  route_key = "GET /api/{version}/requests"
```

Comment:
this looks the same as for adding requests.
why not something unique, like /api/{version}/requests/list ?